system/handlers/formbuilder/item-types/CheckboxList.cfc

```diff
@@ -17,19 +17,24 @@ component {
 	}
 
 	private string function renderResponse( event, rc, prc, args={} ) {
-		var responses = ListToArray( args.response );
-		var values    = ListToArray( args.itemConfiguration.values, chr(13) & chr(10) );
-		var labels    = ListToArray( args.itemConfiguration.labels, chr(13) & chr(10) );
+		var response = args.response ?: "";
 
-		var renderLabels = [];
-		for ( var response in responses ) {
-			var ix = values.find( response );
-			if ( ix>0 ) {
-				renderLabels.append( labels[ix] ?: values[ix] );
+		if ( isEmptyString( response ) ) {
+			return "";
+		}
+
+		var values = ListToArray( args.itemConfiguration.values, chr(13) & chr(10) );
+		var labels = ListToArray( args.itemConfiguration.labels, chr(13) & chr(10) );
+
+		var data = [];
+
+		for ( var i=1; i<=ArrayLen( values ); i++ ) {
+			if ( Find( values[ i ], response ) ) {
```

Comment:
This could cause problems if one value is contained within another. We'll need something more robust here I think. If `response` (`args.response`?) is a comma list, then could do something like:

```cfc
var searchResponse = ",#args.response#,";
// ...
if ( Find( ",#values[ i ]#,", searchResponse ) {
}
```
**Feedback applies throughout the PR**

Reply:
Updated.